api/files/views.py

```diff
@@ -165,3 +169,85 @@ def get_serializer_context(self):
         context = JSONAPIBaseView.get_serializer_context(self)
         context['file'] = self.file
         return context
+
+
+class FileMetadataRecordsList(JSONAPIBaseView, generics.ListAPIView, FileMixin):
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+    )
+
+    required_read_scopes = [CoreScopes.NODE_FILE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = FileMetadataRecordSerializer
+    view_category = 'files'
+    view_name = 'metadata-records'
+
+    ordering = ('-created',)
+
+    def get_queryset(self):
+        return self.get_file().records.all()
+
+
+class FileMetadataRecordDetail(JSONAPIBaseView, generics.RetrieveUpdateAPIView, FileMixin):
+
+    record_lookup_url_kwarg = 'record_id'
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+        PermissionWithGetter(ReadOnlyIfRegistration, 'target'),
+    )
+
+    required_read_scopes = [CoreScopes.NODE_FILE_READ]
+    required_write_scopes = [CoreScopes.NODE_FILE_WRITE]
+
+    serializer_class = FileMetadataRecordSerializer
+    view_category = 'files'
+    view_name = 'metadata-record-detail'
+
+    def get_object(self):
+        return utils.get_object_or_error(
+            self.get_file().records.filter(_id=self.kwargs[self.record_lookup_url_kwarg]),
+            request=self.request,
+        )
+
+
+class FileMetadataRecordDownload(JSONAPIBaseView, generics.RetrieveAPIView, FileMixin):
+
+    record_lookup_url_kwarg = 'record_id'
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+    )
+
+    required_read_scopes = [CoreScopes.NODE_FILE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    view_category = 'files'
+    view_name = 'metadata-record-download'
+
+    def get_serializer_class(self):
+        return None
+
+    def get_object(self):
+        return utils.get_object_or_error(
+            self.get_file().records.filter(_id=self.kwargs[self.record_lookup_url_kwarg]).select_related('schema', 'file'),
+            request=self.request,
+        )
+
+    def get(self, request, **kwargs):
+        file_type = self.request.query_params.get('export', 'json')
+        record = self.get_object()
+        try:
+            response = FileResponse(ContentFile(record.serialize(format=file_type)))
```

Comment:
So I understand, this is where we return the full serialization of the FileMetadataRecord - with all the items pulled off the target/file + user-added metadata items? Only place is a download link?

Reply:
I believe that was the intention, yes! I gotta admit I don't 100% remember the reasoning for this. Adding it to my list of notes of questions to dig up answers to!